src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -0,0 +1,84 @@
+package io.strimzi.kafka.bridge.mqtt.core;
+
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.mqtt.MqttMessage;
+import io.netty.handler.codec.mqtt.MqttMessageBuilders;
+import io.netty.handler.codec.mqtt.MqttMessageType;
+import io.netty.handler.codec.mqtt.MqttPublishMessage;
+import io.netty.handler.codec.mqtt.MqttConnectReturnCode;
+import io.netty.handler.codec.mqtt.MqttConnAckMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.nio.charset.Charset;
+
+/**
+ * Represents a SimpleChannelInboundHandler. The MqttServerHandler is responsible for: <br>
+ * - listen to client connections;<br>
+ * - listen to incoming messages; <br>
+ *
+ * @see io.netty.channel.SimpleChannelInboundHandler
+ */
+public class MqttServerHandler extends SimpleChannelInboundHandler<MqttMessage> {
+    private static final Logger logger = LoggerFactory.getLogger(MqttServerHandler.class);
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) {
+        logger.info("Client  {} is trying to connect", ctx.channel().remoteAddress());
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, MqttMessage msg) {
+        try {
+            MqttMessageType messageType = msg.fixedHeader().messageType();
+            if (messageType == MqttMessageType.CONNECT) {
+                handleConnectMessage(ctx);
+            } else if (messageType == MqttMessageType.PUBLISH) {
+                MqttPublishMessage message = (MqttPublishMessage) msg;
+                message.retain();
+                handlePublishMessage(ctx, message);
+                message.release();
+            } else {
+                // Handle other MQTT message types as needed
+                logger.debug("Got {} message type", messageType.name());
```

Comment:
this should be at the top right after checking the message type, we would log all messages. In this case you are logging everything but CONNECT and PUBLISH.